torch/onnx/_internal/exporter.py

```diff
@@ -155,21 +156,215 @@ def serialize(
         destination.write(export_output.model_proto.SerializeToString())
 
 
+# TODO(bowbao): Add diagnostics for IO formatters.
+@runtime_checkable
+class InputFormatStep(Protocol):
+    def format(
+        self, args: Sequence[Any], kwargs: Mapping[str, Any]
+    ) -> Tuple[Sequence[Any], Mapping[str, Any]]:
+        ...
+
+
+class InputFormatter:
+    """A class that formats the PyTorch model inputs to exported ONNX model inputs format."""
+
+    _input_format_steps: List[InputFormatStep]
+
+    def __init__(self, input_format_steps: Optional[List[InputFormatStep]] = None):
+        self._input_format_steps = input_format_steps or []
+
+    @_beartype.beartype
+    def append_step(self, step: InputFormatStep) -> None:
+        """Appends a step to the input format steps.
+
+        Args:
+            step: The step to append.
+        """
+        self._input_format_steps.append(step)
+
+    @_beartype.beartype
+    def to_onnx(self, *model_args, **model_kwargs) -> Sequence[torch.Tensor]:
+        """Converts the PyTorch model inputs to exported ONNX model inputs format.
+
+        Args:
+            model_args: The PyTorch model inputs.
+            model_kwargs: The PyTorch model keyword inputs.
+
+        Returns:
+            A sequence of tensors converted from PyTorch model inputs.
+        """
+        args: Sequence[Any] = model_args
+        kwargs: Mapping[str, Any] = model_kwargs
+        for step in self._input_format_steps:
+            args, kwargs = step.format(args, kwargs)
+        assert not kwargs
+        return args
+
+
+@runtime_checkable
+class OutputFormatStep(Protocol):
+    def format(self, model_outputs: Any) -> Any:
```

Comment:
Because for return type it is basically `Any`, it could be anything. Your proposal is more restrictive. 
```python
def fn(x):
  return x

def fn_return_args_kwargs(x):
  return [x], {}

output_format_step.format(fn(x))  # doesn't work
output_format_step.format(*fn_return_args_kwargs(x))  # works
```

Reply:
> Because for return type it is basically `Any`, it could be anything. Your proposal is more restrictive.
> 
> ```python
> def fn(x):
>   return x
> 
> def fn_return_args_kwargs(x):
>   return [x], {}
> 
> output_format_step.format(fn(x))  # doesn't work
> output_format_step.format(*fn_return_args_kwargs(x))  # works
> ```

Given
```python
class InputFormatStep(Protocol):
    def format(
        self, model_args: Sequence[Any], model_kwargs: Mapping[str, Any]
    ) -> Tuple[Sequence[Any], Mapping[str, Any]]:
 ```

wouldn't the example above `output_format_step.format(fn(x))` not work too? `x` is not a `Sequence[Any]`  , right?

Reply:
Because function arguments can be written as varargs and varkwargs. Return statement cannot.